src/Entity.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace UKFast\SDK;
+
+abstract class Entity
+{
+    /**
+     * @var array
+     */
+    protected $attributes = [];
+
+    /**
+     * An array of read only attributes
+     * @var array
+     */
+    protected $readOnly = [];
+
+    /**
+     * @param array $attributes
+     */
+    public function __construct($attributes = [])
+    {
+        if (is_object($attributes)) {
+            $attributes = (array) $attributes;
+        }
+        
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * Gets an attribute, if the attribute hasn't been
+     * set yet, return $default
+     *
+     * @param string $attr
+     * @param mixed $default
+     * @return mixed
+     */
+    public function get($attr, $default = null)
+    {
+        if (isset($this->attributes[$attr])) {
+            return $this->attributes[$attr];
+        }
+
+        return $default;
+    }
+
+
+    /**
+     * Sets an attribute
+     *
+     * @param string $attr
+     * @param mixed $value
+     * @return void
+     */
+    public function set($attr, $value)
+    {
+        $this->attributes[$attr] = $value;
+    }
+
+    /**
+     * @return array
+     */
+    public function all()
+    {
+        return $this->attributes;
+    }
+
+    /**
+     * Hydrates an entity
+     *
+     * @param array $attributes
+     * @return void
+     */
+    public function fill($attributes)
+    {
+        foreach ($attributes as $name => $value) {
+            $this->attributes[$name] = $value;
+        }
+    }
+
+    /**
+     * Returns true if the attribute is present on the
```

Comment:
Are we checking the existence of the key or value.

Reply:
It checks if the attribute is present on the Entity. So in this instance, it checks if the key is set